l3build-stdmain.lua

```diff
@@ -28,17 +28,15 @@ local exit   = os.exit
 local insert = table.insert
 
 -- List all modules
-function listmodules()
+local function listmodules()
```

Comment:
I think this needs to be available generally - one might want it in custom `build.lua` code

Reply:
I made it local because it was not documented in `l3build.dtx`.
Is it already used? by LaTeX?
As global function the name is not appropriate, imagine the documentation
`listmodules(): automatically detects module`
instead of
`detectedmodules(): list of automatically detected module`

Decision required.

Reply:
What's wrong with `listmodule()` - it returns a list of modules.